packages/orchestration/src/examples/unbond.flows.js

```diff
@@ -1,38 +1,42 @@
+import { makeTracer } from '@agoric/internal';
+
+const trace = makeTracer('UnbondAndTransfer');
+
 /**
  * @import {Orchestrator, OrchestrationFlow} from '../types.js'
- * @import {TimerService} from '@agoric/time';
- * @import {LocalChain} from '@agoric/vats/src/localchain.js';
- * @import {NameHub} from '@agoric/vats';
- * @import {Remote} from '@agoric/internal';
- * @import {Zone} from '@agoric/zone';
- * @import {CosmosInterchainService} from '../exos/exo-interfaces.js';
- * @import {OrchestrationTools} from '../utils/start-helper.js';
+ * @import {DelegationResponse} from '@agoric/cosmic-proto/cosmos/staking/v1beta1/staking.js';
  */
 
 /**
  * @satisfies {OrchestrationFlow}
  * @param {Orchestrator} orch
  * @param {object} ctx
  * @param {ZCF} ctx.zcf
- * @param {ZCFSeat} _seat
- * @param {undefined} _offerArgs
  */
-export const unbondAndLiquidStake = async (
-  orch,
-  { zcf },
-  _seat,
-  _offerArgs,
-) => {
+export const unbondAndTransfer = async (orch, { zcf }) => {
   console.log('zcf within the membrane', zcf);
   // Osmosis is one of the few chains with icqEnabled
   const osmosis = await orch.getChain('osmosis');
   // In a real world scenario, accounts would be re-used across invokations of the handler
   const osmoAccount = await osmosis.makeAccount();
 
-  // TODO https://github.com/Agoric/agoric-sdk/issues/10016
-  // const delegations = await celestiaAccount.getDelegations();
-  // // wait for the undelegations to be complete (may take weeks)
-  // await celestiaAccount.undelegate(delegations);
+  /** @type {DelegationResponse[]} */
+  const delegations = await osmoAccount.getDelegations();
+  trace('delegations', delegations);
+  // wait for the undelegations to be complete (may take weeks)
+  await osmoAccount.undelegate(
+    delegations.map(d => ({
+      validator: {
+        chainId: 'osmosis-1',
+        value: /** @type {`${string}valoper${string}`} */ (
+          d.delegation.validatorAddress
+        ),
+        encoding: 'bech32',
+      },
+      amount: { denom: 'uosmo', value: BigInt(d.balance.amount) },
+    })),
+  );
```

Comment:
the mismatch between the return of `getDelegations` and the param of `undelegate` is clunky. I'll see about cleaning that up before calling this R4R.

Other feedback welcomed though

Reply:
In 6420537 we moved away from using the `Delegation: {  validatorAddress, delegatorAddress, shares }` type with `undelegate` to better align with the `.delegate()` and `.redelegate()`.

I think this was the correct decision to better align the API, so I would support seeing us massage to `getDelegations: () => { validator: CosmosValidatorAddress, amount: DenomAmount[] }` if that's what you're aiming for.